framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -101,9 +103,23 @@
                 infer(typeFactory, argTypes, assignedTo, methodElem, methodType, targets);
 
         handleUninferredTypeVariables(typeFactory, methodType, targets, inferredArgs);
+        // Type arguments cannot be primitives, so box them if they are.
+        boxPrimitives(typeFactory, inferredArgs);
```

Comment:
`toAtmMap` is called in places other than the return of infer.   I don't think it would be easier to box the primitives there.